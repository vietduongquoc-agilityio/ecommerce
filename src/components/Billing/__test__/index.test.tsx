import React from "react";
import { render } from "@testing-library/react";
import BillingForm from "../";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useSearchParams: jest.fn(),
}));

const mockPush = jest.fn();
describe("BillingForm Snapshot Test", () => {
  beforeEach(() => {
    jest.clearAllMocks();

    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });
  });

  it("renders BillingForm correctly", () => {
    const { asFragment } = render(<BillingForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
