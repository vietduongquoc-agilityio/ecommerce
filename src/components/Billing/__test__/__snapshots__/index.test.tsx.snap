// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingForm Snapshot Test renders BillingForm correctly 1`] = `
<DocumentFragment>
  <section
    style="display: flex; width: 100%; max-width: 1245px; margin: 0px auto; justify-content: space-between; padding: 100px 0px;"
  >
    <aside
      style="display: flex; flex-direction: column; width: 100%; max-width: 455px;"
    >
      <h2
        style="font-weight: 600; font-size: 36px; color: rgb(0, 0, 0); margin-bottom: 37px;"
      >
        Billing details
      </h2>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <div>
          <label
            class="labelInput"
          >
            First Name
          </label>
          <input
            class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
            placeholder="Enter..."
            style="margin-bottom: 23px; margin-top: 23px;"
            type="text"
          />
        </div>
        <div>
          <label
            class="labelInput"
          >
            Last Name
          </label>
          <input
            class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
            placeholder="Enter..."
            style="margin-bottom: 23px; margin-top: 23px;"
            type="text"
          />
        </div>
      </div>
      <label
        class="labelInput"
      >
        Country / Region
      </label>
      <select
        class="contentInput"
        id="Country"
        name="Country"
      >
        <option
          value="VietNam"
        >
          Viet Nam
        </option>
        <option
          value="USA"
        >
          USA
        </option>
        <option
          value="China"
        >
          China
        </option>
        <option
          value="Japan"
        >
          Japan
        </option>
      </select>
      <label
        class="labelInput"
      >
        Street address
      </label>
      <input
        class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
        placeholder="Enter..."
        type="text"
      />
      <label
        class="labelInput"
      >
        Town / City
      </label>
      <input
        class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
        placeholder="Enter..."
        type="text"
      />
      <label
        class="labelInput"
      >
        ZIP code
      </label>
      <input
        class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
        placeholder="Enter..."
        type="text"
      />
      <label
        class="labelInput"
      >
        Phone
      </label>
      <input
        class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
        placeholder="Enter..."
        type="text"
      />
      <label
        class="labelInput"
      >
        Email address
      </label>
      <input
        class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black contentInput"
        placeholder="Enter..."
        type="text"
      />
    </aside>
    <aside
      style="width: 527px;"
    >
      <div
        style="display: flex; justify-content: space-between; font-weight: 500; font-size: 24px; margin-bottom: 22px;"
      >
        <p>
          Product
        </p>
        <p>
          Subtotal
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between; margin-bottom: 22px;"
      >
        <p
          style="font-weight: 400; font-size: 16px; color: rgb(159, 159, 159);"
        >
          Asgaard sofa x 1
        </p>
        <p
          style="font-weight: 300px; font-size: 16px;"
        >
          Rs. 250,000.00
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between; margin-bottom: 22px;"
      >
        <p
          style="font-weight: 400; font-size: 16px; color: rgb(159, 159, 159);"
        >
          Subtotal
        </p>
        <p
          style="font-weight: 300px; font-size: 16px;"
        >
          Rs. 250,000.00
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between; border-bottom: 1px solid #D9D9D9;"
      >
        <p
          style="font-weight: 400; font-size: 16px; color: rgb(159, 159, 159);"
        >
          Subtotal
        </p>
        <p
          style="font-weight: 700; font-size: 24px; color: rgb(184, 142, 47); margin-bottom: 25px;"
        >
          Rs. 520,000.00
        </p>
      </div>
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <p
          style="font-weight: 300; font-size: 16px; margin-top: 56px; margin-bottom: 26px;"
        >
          Your personal data will be used to support your experience throughout this website, to manage access to your account, and for other purposes described in our privacy policy.
        </p>
        <button
          style="padding: 18px 102px; font-size: 18px; font-weight: 600; text-align: center; width: 318px; height: auto; cursor: pointer; transition: all 0.3s ease; border: 1px solid #000000; background-color: transparent; color: rgb(0, 0, 0); border-radius: 15px;"
        >
          Place order
        </button>
      </div>
    </aside>
  </section>
</DocumentFragment>
`;
